### DISCUSSION TITLE
Idea for CIP - Relax rules requiring a user to have published a pubkey

### DISCUSSION THREAD

**Discussion by B0BSmiths** [CounterpartyXCP/CIPs#125] [CIPs]
Now that Counterparty mainly uses OP_RETURN it would make on boarding new users easier if the rule requiring a user to have previously published a pubkey could be relaxed

A pub key is only required when multisig encoding, without burn keys, is to be used, operations such as sending an asset, opening a dispenser, closing a dispenser, destroying an asset, even posting a short op_returnable broadcast do not require a pub key,  yet users are prevented from using these features by the CP API requiring a pubkey it does not even need

Ideally only operations that require a pubkey should enforce the old logic

**Comment by jdogresorg** [comment]
I agree, we should not require the public key unless we actually USE the public key (multisig encoding).

If you wanted to make this a github issue in the counterparty-lib repo, I can work to get this unnecessary requirement removed in the next release.

**Comment by B0BSmiths** [comment]
My apologies, I think it is already the case it only requires pubkey when it actually requires it, I did some more testing and I was able to create a broadcast with a burnkey without having transacted before.
